examples/bri-3/src/bri/identity/bpiSubjects/models/bpiSubject.interface.ts

```diff
@@ -1,5 +1,6 @@
 import { BpiSubjectType } from "./bpiSubjectType.enum";
 
+// TODO: Do we actually need interfaces as we are already working with classes?
```

Comment:
@ognjenkurtic If you want to create and pass a type-checked class object, you should use ts classes. If you need to work without creating an object, an interface is best.

Use an interface for the following:
- Validating specific structure of properties
- Objects as parameters
- Objects returned from functions

So we need to determine if we want to use classes but not instantiate them, then we combine interfaces with classes.

Get's off his soapbox

Reply:
to me now these interfaces seem redundant, could change i guess

Reply:
Interfaces are redundant for me as well. We want to instantiate objects and these objects will encapsulate all the logic for their state changing. Let's discuss in a separate issue, not a blocker for now.